internal/backends/sqlite/metadata/registry.go

```diff
@@ -353,6 +374,94 @@ func (r *Registry) CollectionRename(ctx context.Context, dbName, oldCollectionNa
 	panic("not implemented")
 }
 
+// indexesCreate creates an index in the collection.
+//
+// It does not hold the lock.
+func (r *Registry) indexesCreate(ctx context.Context, db *fsql.DB, c *Collection, indexes []IndexInfo) error {
+	err := db.InTransaction(ctx, func(tx *fsql.Tx) error {
+		for _, index := range indexes {
+			// check if index already exists
+			q := `SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type= 'index' and tbl_name = ? and name = ?)`
+			row := tx.QueryRowContext(ctx, q, c.TableName, c.TableName+index.Name)
+
+			var exists bool
+			err := row.Scan(&exists)
+			if err != nil {
+				return lazyerrors.Error(err)
+			}
+
+			if exists {
+				continue
+			}
+
+			q = "CREATE "
+
+			if index.Unique {
+				q += "UNIQUE "
+			}
+
+			q += "INDEX %q ON %q (%s)"
+
+			columns := make([]string, len(index.Key))
+			for i, key := range index.Key {
+				columns[i] = fmt.Sprintf("%s->'$.%s'", DefaultColumn, key.Field)
+				if key.Descending {
+					columns[i] += " DESC"
+				}
+			}
+
+			q = fmt.Sprintf(q, c.TableName+index.Name, c.TableName, strings.Join(columns, ", "))
```

Comment:
Could we use a placeholder `?` or `$` for fields? I suppose `index.Name` can be user input.

Reply:
Yeah, it can be user input that breaks everything (there is  `DangerousKey` test that currently fails). I tried `?` but I didn't find an approach that would work.

Reply:
I think this one would be important one. I wonder why :thinking:

Reply:
There should be a better way to escape this string anyway... I'll take a look at how we do it for pg (there was sort of a hack to do it), maybe we can do smth similar here.